src/Sleep.js

```diff
@@ -4,78 +4,104 @@ class Sleep {
   }
+
   calculateDailySleepQuality(id, date) {
-    let findSleepQualityByDate = this.sleepData.find((data) => id === data.userID && date === data.date);
+    let findSleepQualityByDate = this.findSleepDataByDate(id, date);
     return findSleepQualityByDate.sleepQuality;
   }
+
   calculateWeekSleep(date, id, userRepo) {
     return userRepo.getWeekFromDate(date, id, this.sleepData).map((data) => `${data.date}: ${data.hoursSlept}`);
   }
+
   calculateWeekSleepQuality(date, id, userRepo) {
     return userRepo.getWeekFromDate(date, id, this.sleepData).map((data) => `${data.date}: ${data.sleepQuality}`);
   }
+
   calculateAllUserSleepQuality() {
-    var totalSleepQuality = this.sleepData.reduce(function(sumSoFar, dataItem) {
-      sumSoFar += dataItem.sleepQuality;
-      return sumSoFar;
-    }, 0)
-    return totalSleepQuality / sleepData.length
+    var totalSleepQuality = this.getAverageOfValues(this.sleepData, "sleepQuality");
+    return totalSleepQuality;
   }
-  determineBestSleepers(date, userRepo) {
+
+  getUserSleepObject(date, userRepo, sleepProperty) {
     let timeline = userRepo.chooseWeekDataForAllUsers(this.sleepData, date);
-    let userSleepObject = userRepo.isolateUsernameAndRelevantData(this.sleepData, date, 'sleepQuality', timeline);
-
-    return Object.keys(userSleepObject).filter(function(key) {
-      return (userSleepObject[key].reduce(function(sumSoFar, sleepQualityValue) {
-        sumSoFar += sleepQualityValue
-        return sumSoFar;
-      }, 0) / userSleepObject[key].length) > 3
-    }).map(function(sleeper) {
+    let userSleepObject = userRepo.combineRankedUserIDsAndAveragedData(this.sleepData, date, sleepProperty, timeline);
+    return userSleepObject;
+  }
+
+  getUserNames(userSleepObject, userRepo) {
+    return Object.keys(userSleepObject).filter((key) => {
+      return this.getAverageOfValues(userSleepObject[key]) > 3;
+    }).map((sleeper) => {
       return userRepo.getDataFromID(parseInt(sleeper)).name;
     })
   }
-  determineSleepWinnerForWeek(date, userRepo) {
+
+  determineBestSleepers(date, userRepo) {
     let timeline = userRepo.chooseWeekDataForAllUsers(this.sleepData, date);
-    let sleepRankWithData = userRepo.combineRankedUserIDsAndAveragedData(this.sleepData, date, 'sleepQuality', timeline);
+    let userSleepObject = userRepo.isolateUsernameAndRelevantData(this.sleepData, date, "sleepQuality", timeline);
+    return this.getUserNames(userSleepObject, userRepo)
+  }
 
+  determineSleepWinnerForWeek(date, userRepo) {
+    let sleepRankWithData = this.getUserSleepObject(date, userRepo, "sleepQuality");
     return this.getWinnerNamesFromList(sleepRankWithData, userRepo);
   }
-  determineSleepHoursWinnerForDay(date, userRepo) {
-    let timeline = userRepo.chooseDayDataForAllUsers(this.sleepData, date);
-    let sleepRankWithData = userRepo.combineRankedUserIDsAndAveragedData(this.sleepData, date, 'hoursSlept', timeline);
 
+  determineSleepHoursWinnerForDay(date, userRepo) {
```

Comment:
Great split the old function. It played too many different roles before, it makes more sense to have a separate method for each variable.